pkg/controller/subnet.go

```diff
@@ -1356,6 +1356,10 @@ func (c *Controller) checkGwNodeExists(gatewayNode string) bool {
 
 func (c *Controller) deleteEcmpRouteForNode(subnet *kubeovnv1.Subnet) error {
 	for _, gw := range strings.Split(subnet.Spec.GatewayNode, ",") {
+		if subnet.Status.ActivateGateway != "" && subnet.Status.ActivateGateway == gw {
+			continue
+		}
+
```

Comment:
https://github.com/kubeovn/kube-ovn/wiki/%E9%9B%86%E4%B8%AD%E5%BC%8F%E5%AD%90%E7%BD%91%E7%BD%91%E5%85%B3NAT%E6%A8%A1%E5%BC%8F%E6%8C%87%E5%AE%9A%E5%87%BA%E7%BD%91IP

In this case, gw is not equal to subnet.Status.ActivateGateway. 
Check condition not to be met is a good suggestion, and the check can be moved after gw is valued.

Reply:
ok